src/ApparentHorizons/StrahlkorperInDifferentFrame.cpp

```diff
@@ -56,14 +57,27 @@ void coords_to_different_frame(
     get<2>(x_src) = get<2>(src_cartesian_coords)[s];
     for (const auto& block : domain.blocks()) {
       const auto& logical_to_grid_map = block.moving_mesh_logical_to_grid_map();
-      const auto x_logical = logical_to_grid_map.inverse(x_src);
-      // x_logical might be an empty std::optional.
-      if (x_logical.has_value() and get<0>(x_logical.value()) <= 1.0 and
-          get<0>(x_logical.value()) >= -1.0 and
-          get<1>(x_logical.value()) <= 1.0 and
-          get<1>(x_logical.value()) >= -1.0 and
-          get<2>(x_logical.value()) <= 1.0 and
-          get<2>(x_logical.value()) >= -1.0) {
+      auto x_logical = logical_to_grid_map.inverse(x_src);
```

Comment:
Can you call `block_logical_coordinates(domain, src_cartesian_coords, time, functions_of_time)` at the beginning of the function and use the result for the computation? That's what's being done in `StrahlkorperCoordsInDifferentFrame.cpp:63`.

Reply:
What's actually the difference between the two?

Reply:
No because have no notion of elements here. This is just a bunch of random points on a sphere. block_logical_coords returns vectors of elements, but we need to check point-by-point which block the point is in so we get the correct maps

Reply:
No, block_logical_coordinates returns a vector of size num_points, and gives you the block ID and the logical coords within that block for each input coordinate.